usaspending_api/etl/management/commands/load_table_from_delta.py

```diff
@@ -164,44 +176,184 @@ def handle(self, *args, **options):
+
+            # Create a Dataframe with a unique list of entities (based on the table's unique key) that have been
+            # updated since this command was last run. If an entity has been updated multiple times, the latest
+            # update will be selected. This utilizes Delta Lake's Change Data Feed feature.
+            # This will be used to determine if the number of updated records exceeds the update threshold, and
+            # (if not exceeding) it will serve as the basis for dataframes to be written to Postgres with incremental
+            # changes.
+            unique_identifiers = self.table_spec["unique_identifiers"]
+            distinct_df = spark.sql(
+                f"""
+                SELECT * FROM (
+                    SELECT * ,
+                    ROW_NUMBER() OVER (PARTITION BY {', '.join(unique_identifiers)} ORDER BY _commit_version DESC) AS row_num
+                    FROM table_changes('{self.qualified_source_delta_table}', {last_live_version + 1})
+                    WHERE _change_type in ('insert', 'update_postimage', 'delete')
+                ) WHERE row_num = 1
+            """
+            )
+
+            if not self._surpassed_update_threshold(spark, distinct_df):
+                upsert_table_suffix = "temp_upserts"
+                delete_table_suffix = "temp_deletes"
+
+                # Upsert Schema (matches base table)
+                qualified_upsert_temp_table = self._prepare_temp_table(postgres_temp_schema, upsert_table_suffix)
+
+                # Delete Schema
+                delete_schema_def = self.table_spec["incremental_delete_temp_schema"]
+                qualified_delete_temp_table = self._prepare_temp_table(
+                    postgres_temp_schema, delete_table_suffix, schema_override=delete_schema_def
+                )
+
+                # Split the Dataframe into two based on the type of update that was made
+                upsert_df = distinct_df.filter("_change_type IN ('insert', 'update_postimage')")
+                delete_df = distinct_df.filter("_change_type IN ('delete')")
+
+                self._write_to_postgres(
+                    spark, upsert_df, qualified_upsert_temp_table, postgres_schema_def, self.table_spec["column_names"]
+                )
+                self._write_to_postgres(
+                    spark, delete_df, qualified_delete_temp_table, delete_schema_def, list(delete_schema_def)
+                )
+
+                updated_incrementally = True
+
+        if not updated_incrementally:
+            temp_table_suffix = "temp"
+
+            qualified_temp_table = self._prepare_temp_table(postgres_temp_schema, temp_table_suffix)
+
+            # Read from Delta
+            df = spark.table(source_delta_table)
+
+            # Reset the sequence before load for a table if it exists
+            if self.options["reset_sequence"] and self.table_spec.get("postgres_seq_name"):
```

Comment:
Is `self.options["reset_sequence"]` a Boolean? If not I think this if statement could be made more clear.

Reply:
Yes, this is a boolean